internal/internal_poller_autoscaler_test.go

```diff
@@ -190,9 +193,9 @@ func Test_pollerAutoscaler(t *testing.T) {
 				go func() {
 					defer wg.Done()
 					for pollResult := range pollChan {
-						pollerScaler.Acquire(1)
+						pollerScaler.permit.Acquire(context.Background())
```

Comment:
`Acquire` returns error which should be handled. otherwise `Release` might be called without acquire succeeding which leads to panic

Reply:
Good catch. I have been searching for the issue for hours.